File: 1-Basics/1.1-Patterns/Patterns_python/18-AlphaHill.py
n = int(input("Enter Row Size N : "))


for i in range(1,n+1):
    c = 'A'
    # space
    for j in range(n-i):
        print(" ", end="")
    # left
    for k in range(i):
        print(c, end="")
        c = chr(ord(c)+1)
    # right
    for l in range(i-1,0,-1):
        print(chr(ord('A')+l-1), end="")
    
    print('\n')

File: 1-Basics/1.1-Patterns/Patterns_cpp/18-AlphaHill.cpp
#include<bits/stdc++.h>
using namespace std ;

// ---A
// --ABA
// -ABCBA
// ABCDCBA

void pattern(int n){
    for(int i=1; i<=n; i++){
        char c = 'A';
        
        // space
        for(int j=n-i; j>0; j--){
            cout<<" ";
        }

        // left
        for(int k=1; k<=i; k++){
            cout<<c;
            c++;
        }

        // right
        for(char l='A'+i-2; l>='A'; l--){
            cout<<l;
        }

        cout<<endl;
    }
}

int main(){
    cout<<"Enter Row Size N : ";
    int n;
    cin>>n;

    pattern(n);

    return 0;
}